source/ualloc.c

```diff
@@ -47,12 +52,29 @@ const char ua_chars[] = "NFEWIL";
     #define caller_addr() (NULL)
 #endif
 
+/**
+ * Allocate memory with traits
+ *
+ * ualloc allocates memory with traits. Traits are a way of specifying additional behaviour to the allocator.
+ * Currently, ualloc supports a limited number of traits:
+ * * UALLOC_TRAITS_NEVER_FREE: Allocate memory with krbs, which is interrupt-safe
+ * * UALLOC_TRAITS_ZERO_FILL:  memset the returned memory to 0.
+ * These traits can be used on their own or together.
+ *
+ * @param[in] bytes the number of bytes to allocates
+ * @param[in] traits the traits to apply to the allocated memory
+ * @return a pointer to the allocated memory on success, or NULL on failure.
+ */
 void * mbed_ualloc(size_t bytes, UAllocTraits_t traits)
 {
     void * ptr = NULL;
     void * caller = (void*) caller_addr();
     if (UALLOC_TEST_TRAITS(traits.flags, UALLOC_TRAITS_NEVER_FREE)) {
         ptr = mbed_krbs(bytes);
+        // krbs uses the same semantics as sbrk, so translate a -1 to NULL.
+        if (ptr == -1) {
```

Comment:
No warnings on this line? I would've expected `ptr == (void*)-1` or something similar.